wicket-guice/src/main/java/org/apache/wicket/guice/GuiceFieldValueFactory.java

```diff
@@ -64,15 +68,34 @@ public Object getFieldValue(final Field field, final Object fieldOwner)
 				{
 					boolean optional = injectAnnotation != null && injectAnnotation.optional();
 					Annotation bindingAnnotation = findBindingAnnotation(field.getAnnotations());
-					final IProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+					final GuiceProxyTargetLocator locator = new GuiceProxyTargetLocator(field, bindingAnnotation, optional);
+
+					Object cachedValue = cache.get(locator);
```

Comment:
It's not, but i'm not sure about the objensis performance, and the spring ioc also caches the proxies, so it keeps those 2 closer together aswel